code/modules/surgery/organs/organ_internal.dm

```diff
@@ -0,0 +1,314 @@
+#define PROCESS_ACCURACY 10
+
+/obj/item/organ/internal
+	origin_tech = "biotech=2"
+	force = 1
+	w_class = 2
+	throwforce = 0
+	var/zone = "chest"
+	var/slot
+	vital = 0
+	var/organ_action_name = null
+
+/obj/item/organ/internal/proc/insert(mob/living/carbon/M, special = 0)
+	if(!iscarbon(M) || owner == M)
+		return
+
+	var/obj/item/organ/internal/replaced = M.get_organ_slot(slot)
+	if(replaced)
+		replaced.remove(M, special = 1)
+
+	owner = M
+	M.internal_organs |= src
+	loc = null
+	if(organ_action_name)
+		action_button_name = organ_action_name
+
+
+/obj/item/organ/internal/proc/remove(mob/living/carbon/M, special = 0)
+	owner = null
+	if(M)
+		M.internal_organs -= src
+		if(vital && !special)
+			M.death()
+
+	if(organ_action_name)
+		action_button_name = null
+
```

Comment:
You'll want to remove its entry from "internal_organs_by_name" here, too, if it's the organ that corresponds to that tag (otherwise you get IPC multibrain shenanigans and stuff like that!)